### PR TITLE
docs: glossary infra

### LINKED ISSUE #33117: docs: setup glossary
This bounty is for both setting up the glossary infrastructure and populating it with most openpilot terms.

Requirements:
* a master definition file
* "glossary page" under the concepts section
* nice to have: click or hover on words defined in the glossary to get the definition on any docs.comma.ai page. bounty value doubles if this is done

wiki has a good starting point on definitions: https://github.com/commaai/openpilot/wiki/General-Terms

### PR DESCRIPTION
**Pull request by LouisVelez**
This PR replaces #34155 and splits the work needed for the glossary infra. This PR intention is to only set-up the infrastructure per the bounty specifications, including hover over words

Remaining work:
- [x] Add term file #34232 
- [x]  Modify`glossary.md`#34234

Once This PR and the term file gets merged the modified `glossary.md` should be good to go. 

Closes #33117

### CODE REVIEW

**Comment by LouisVelez** [comment]
@adeebshihadeh! In the original PR, you requested that I revert this change ([see comment](https://github.com/commaai/openpilot/pull/34155/files#r1881076080)). I had made that adjustment to prevent the CI from failing and to keep the build process simple. The alternatives would be either adding steps to use Python 3.11+ (which includes tomllib by default) or relying on a third-party package, though the latter isn’t my preferred approach. 

Please provide some insight on how to proceed.

**Comment by adeebshihadeh** [comment]
Oh, that's fine then. I thought `ubuntu-latest` pointed to 24.04 nowadays anyway.

**Comment by LouisVelez** [comment]
> Oh, that's fine then. I thought `ubuntu-latest` pointed to 24.04 nowadays anyway.

No problem, It seems the rollout started on Dev 5th and they plan on it being completed by Jan 17th. [Source](https://github.com/actions/runner-images/issues/10636)

File: docs/hooks/glossary.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import os
+import re
+import tomllib
+from typing import Any
+
+def load_glossary(file_path="docs/glossary.toml"):
+  if not os.path.exists(file_path):
+    return {}
```
Comment (adeebshihadeh) [inline docs/hooks/glossary.py:8]: this should error, not fail silently. for this PR, you can just include an empty one or port over the existing glossry

File: docs/hooks/glossary.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  with open(file_path, "rb") as f:
+    glossary_data = tomllib.load(f)
+  return glossary_data.get("glossary", {})
+
+glossary: dict[str, Any] | None = None
```
Comment (adeebshihadeh) [inline docs/hooks/glossary.py:13]: do we need a global?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: docs/hooks/glossary.py (hunk 1 above)
Comment (LouisVelez) [reply docs/hooks/glossary.py:8]: Will add an empty. I made a separate PR for the glossary term file #34232

File: docs/hooks/glossary.py (hunk 2 above)
Comment (LouisVelez) [reply docs/hooks/glossary.py]: taken care of with latest push

File: docs/hooks/glossary.py (hunk 1 above)
Comment (LouisVelez) [reply docs/hooks/glossary.py]: pushed

**Pull request merged**